### DISCUSSION TITLE
Remove Kindle verbiage?

### DISCUSSION THREAD

**Discussion by benisek00** [zenangst/KeyboardCowboy#516] [General]
Every time I copy and paste from a Kindle book to a TextEdit file where I keep notes and important points, it also pastes "Book Title - XYZ Publishing. Kindle Edition.". Is there a way to program KeyboardCowboy with that exact text so that every time I copy-paste, what's added by the Kindle is automatically removed from the TextEdit file?

**Comment by zenangst** [comment]
Hey @benisek00, I guess you could pipe it out to a shell script, strip the added text and then perform an action in the app that you want.

Are you familiar with shell scripting?

**Comment by benisek00** [comment]
Hello!
Thank you for your email! 
Does that answer your question if I tell you that I have no idea what "pipe out” even means?
Sorry - I’m probably not the target audience for your app - but I love it, I use it for signatures and short bits of texts in my emails.
:)

**Comment by zenangst** [comment]
@benisek00 I didn't test this out myself but I guess something like this could work:

<img width="1077" alt="image" src="https://github.com/zenangst/KeyboardCowboy/assets/57446/c4d2474c-fc70-4dc5-ad89-09d4af6388ed">

Just to explain what it does.

`pbpaste` gets the current contents of your pasteboard.
We save that into a variable called `clipboard_content`.
We then try and remove the string that you provided using `sed`.

`Book Title - XYZ Publishing. Kindle Edition.`

Finally, to wrap things up, we take our updated content and send it back to the pasteboard using `pbcopy`.

**Comment by benisek00** [comment]
You are fantastic. Will try it right now!
